src/Common/test/Certificates.Test/ConfigureCertificateOptionsTest.cs

```diff
@@ -17,176 +18,287 @@ public sealed class ConfigureCertificateOptionsTest
-        var configureOptions = new ConfigureCertificateOptions(configurationRoot);
+        var appSettings = new Dictionary<string, string?>
+        {
+            [$"{GetConfigurationKey(certificateName, "CertificateFilePath")}"] = "instance.crt",
+            [$"{GetConfigurationKey(certificateName, "PrivateKeyFilePath")}"] = "instance.key"
+        };
+
+        IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(appSettings).Build();
+        var configureOptions = new ConfigureCertificateOptions(configuration);
         var options = new CertificateOptions();
 
-        configureOptions.Configure(CertificateName, options);
+        configureOptions.Configure(certificateName, options);
 
         options.Certificate.Should().NotBeNull();
         options.Certificate.HasPrivateKey.Should().BeTrue();
     }
 
-    [Fact]
-    public async Task CertificateOptionsUpdateOnFileContentChange()
+    [Theory]
+    [InlineData("")]
+    [InlineData(CertificateName)]
+    public async Task CertificateOptionsUpdateOnFileContentChange(string certificateName)
     {
         using var sandbox = new Sandbox();
         string firstCertificateContent = await File.ReadAllTextAsync("instance.crt", TestContext.Current.CancellationToken);
         string firstPrivateKeyContent = await File.ReadAllTextAsync("instance.key", TestContext.Current.CancellationToken);
-        var firstX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
-        string secondCertificateContent = await File.ReadAllTextAsync("instance2.crt", TestContext.Current.CancellationToken);
-        string secondPrivateKeyContent = await File.ReadAllTextAsync("instance2.key", TestContext.Current.CancellationToken);
-        var secondX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
-        string certificateFilePath = sandbox.CreateFile("cert", firstCertificateContent);
-        string privateKeyFilePath = sandbox.CreateFile("key", firstPrivateKeyContent);
-
+        using var firstX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
+        string secondCertificateContent = await File.ReadAllTextAsync("secondInstance.crt", TestContext.Current.CancellationToken);
+        string secondPrivateKeyContent = await File.ReadAllTextAsync("secondInstance.key", TestContext.Current.CancellationToken);
+        using var secondX509 = X509Certificate2.CreateFromPemFile("secondInstance.crt", "secondInstance.key");
+        string certificateFilePath = sandbox.CreateFile(Guid.NewGuid() + ".crt", firstCertificateContent);
+        string privateKeyFilePath = sandbox.CreateFile(Guid.NewGuid() + ".key", firstPrivateKeyContent);
+
+        MemoryFileProvider fileProvider = new();
+        fileProvider.IncludeFile(MemoryFileProvider.DefaultAppSettingsFileName, BuildAppSettingsJson(certificateName, certificateFilePath, privateKeyFilePath));
```

Comment:
Please extract the result from `BuildAppSettingsJson` into a variable. It is impossible to debug this way.